tests/test_group.rs

```diff
@@ -17,97 +18,149 @@ fn create_commit_optional_path() {
+    let bob_signature_key = &bob_identity.get_signature_key_pair().get_private_key();
+
+    // Mandatory extensions
+    let capabilities_extension = Box::new(CapabilitiesExtension::default());
+    let lifetime_extension = Box::new(LifetimeExtension::new(60));
+    let mandatory_extensions: Vec<Box<dyn Extension>> =
+        vec![capabilities_extension, lifetime_extension];
+
     // Generate KeyPackages
     let alice_key_package_bundle = KeyPackageBundle::new(
         ciphersuite_name,
-        &alice_identity.get_signature_key_pair().get_private_key(), // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
-        Credential::from(MLSCredentialType::Basic(alice_credential.clone())), // TODO: this consumes the credential!
-        Vec::new(),
+        alice_signature_key,
+        Credential::from(MLSCredentialType::Basic(alice_credential.clone())),
+        mandatory_extensions.clone(),
     );
 
     let bob_key_package_bundle = KeyPackageBundle::new(
         ciphersuite_name,
-        &bob_identity.get_signature_key_pair().get_private_key(), // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
-        Credential::from(MLSCredentialType::Basic(bob_credential)), // TODO: this consumes the credential!
-        Vec::new(),
+        bob_signature_key,
+        Credential::from(MLSCredentialType::Basic(bob_credential)),
+        mandatory_extensions.clone(),
     );
     let bob_key_package = bob_key_package_bundle.get_key_package();
 
     let alice_update_key_package_bundle = KeyPackageBundle::new(
         ciphersuite_name,
-        &alice_identity.get_signature_key_pair().get_private_key(), // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
+        alice_signature_key,
         Credential::from(MLSCredentialType::Basic(alice_credential)),
-        Vec::new(),
+        mandatory_extensions,
     );
     let alice_update_key_package = alice_update_key_package_bundle.get_key_package();
+    assert!(alice_update_key_package.verify().is_ok());
 
     // Alice creates a group
     let group_id = [1, 2, 3, 4];
-    let group_alice_1234 = MlsGroup::new(&group_id, ciphersuite_name, alice_key_package_bundle);
+    let mut group_alice_1234 = MlsGroup::new(&group_id, ciphersuite_name, alice_key_package_bundle);
 
-    // Alice adds Bob
+    // Alice adds Bob with forced self-update
     let bob_add_proposal = group_alice_1234.create_add_proposal(
         group_aad,
-        &alice_identity.get_signature_key_pair().get_private_key(),
+        alice_signature_key,
         bob_key_package.clone(),
     );
+    let epoch_proposals = vec![bob_add_proposal];
+    let (mls_plaintext_commit, _welcome_bundle_alice_bob_option) = match group_alice_1234
+        .create_commit(group_aad, alice_signature_key, epoch_proposals, true)
```

Comment:
This more about the API. But having boolean arguments, especially in public APIs, is not great. It's never clear what the `true` here means. So for this API this should be something like `true /* force self-update */`. But maybe we should consider adding a separate function with this set to `true`, like `create_commit_self_update`.